components/reusable-components/reusable-table/TableRowComponent.tsx

```diff
@@ -10,46 +10,83 @@ interface TableRowComponentProps<T extends { id: string }> {
   onCheckboxChange: (id: string) => void;
   showCheckbox?: boolean;
   renderActions?: (item: T) => React.ReactNode;
-  renderActionsDropdown?: ReactNode;
+  renderActionsDropdown?: React.ReactNode;
+  onClick?: (id: string) => void;
 }
 
 const TableRowComponent = <T extends { id: string }>({
   data,
   displayFields,
   isChecked,
   onCheckboxChange,
-  showCheckbox,
+  showCheckbox = false,
   renderActions,
   renderActionsDropdown,
+  onClick,
 }: TableRowComponentProps<T>) => {
   const handleCheckboxChange = () => {
     onCheckboxChange(data.id);
   };
 
+  const handleRowClick = () => {
+    if (onClick) {
+      onClick(data.id);
+    }
+  };
+
+  const handleActionClick = (event: React.MouseEvent) => {
+    event.stopPropagation(); // Prevent row click from being triggered
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter' || event.key === ' ') {
+      handleActionClick(event as any);
+    }
+  };
+
   return (
-    <tr className={style.rowComponent}>
+    <tr
+      className={style.rowComponent}
+      onClick={handleRowClick} // Handle row click
+      style={{ cursor: onClick ? 'pointer' : 'default' }}
+    >
       {showCheckbox && (
         <td aria-label="Checkbox">
           <input type="checkbox" checked={isChecked} onChange={handleCheckboxChange} />
         </td>
       )}
       {displayFields.map((field) => (
-        <td key={field as string}>
+        <td key={String(field)}>
           {Array.isArray(data[field])
-            ? (data[field] as unknown as { name: string }[]).map((item) => item.name).join(', ')
+            ? (data[field] as { name: string }[]).map((item) => item.name).join(', ')
             : String(data[field])}
         </td>
       ))}
-
       {renderActionsDropdown && (
         <td className={style.actionCell} aria-label="Actions">
-          {renderActionsDropdown}
+          <button
+            type="button"
+            onClick={handleActionClick}
+            onKeyDown={handleKeyDown}
+            tabIndex={0} // Make button focusable
+            aria-haspopup="true" // Indicate a dropdown menu
+          >
+            {renderActionsDropdown}
+          </button>
         </td>
       )}
-
       {renderActions && (
         <td className={style.actionCell}>
-          <div className={style.actionButtons}>{renderActions(data)}</div>
+          <button
+            type="button"
+            className={style.actionButtons}
+            onClick={handleActionClick}
+            onKeyDown={handleKeyDown}
+            tabIndex={0} // Make button focusable
+            aria-haspopup="true" // Indicate a dropdown menu
+          >
+            {renderActions(data)}
+          </button>
```

Comment:
You have two nested buttons here. 
<img width="127" alt="Screenshot 2024-09-08 at 15 20 39" src="https://github.com/user-attachments/assets/b04d0a4b-21a8-4b46-a4ff-2596ad7fddd2">
Please revert the changes you did in the ActionDropdown. You can pass onClick function and also styles if you need to overwrite the default styles we already have.